pkg/server/server.go

```diff
@@ -41,14 +35,20 @@ func StartAndListen(playmode bool) {
 		TLSNextProto: make(map[string]func(*http.Server, *tls.Conn, http.Handler)),
 	}
 
-	router.Handle("/searchapi/graphql", playground.Handler("GraphQL playground", "/query"))
-	router.Handle("/query", srv.Handler)
-	if playmode {
-		klog.Infof("connect to https://localhost:%d%s/graphql for GraphQL playground", port, config.Cfg.ContextPath)
-		klog.Fatal(http.ListenAndServeTLS(":"+fmt.Sprint(port), "./sslcert/tls.crt", "./sslcert/tls.key",
-			router))
+	// Initiating router
+	router := mux.NewRouter()
+
+	if config.Cfg.PlaygroundMode {
+		router.Handle("/playground", playground.Handler("GraphQL playground", "/searchapi/graphql"))
+		klog.Infof("GraphQL playground is now running on https://localhost:%d/playground", port)
 	}
+
+	// Add authentication middleware to the /searchapi subroute.
```

Comment:
- Always serve the route `/searchapi/graphql`
- The Playground route `/playground` is optional.
- Only add authentication middleware to `/searchapi/graphql`